ts/components/conversation/composition/CompositionInput.tsx

```diff
@@ -0,0 +1,861 @@
+          }
+        }
+      },
+      [handleUndo, createSyntheticEvent, handleRedo]
+    );
+
+    useEffect(() => {
+      const el = elRef.current;
+      if (!el) {
+        return;
+      }
+      el.addEventListener('beforeinput', handleBeforeInput);
+      // eslint-disable-next-line consistent-return
+      return () => el.removeEventListener('beforeinput', handleBeforeInput);
+    }, [elRef, handleBeforeInput]);
+
+    // Update DOM on html change
+    useLayoutEffect(() => {
+      /**
+       * After first edit the input html will minimally contain only a br tag, so even if the user deletes
+       * everything the html will just be a single br tag. If the value is just one br tag we want to remove
+       * it so the input is empty and the placeholder appears.
+       */
+      const el = elRef.current;
+      if (!el) {
+        return;
+      }
+
+      const normalizedHtml = normalizeHtml(html);
+      const normalizedCurrentHtml = normalizeHtml(el.innerHTML);
+
+      if (normalizedHtml.includes('<br>')) {
+        window.log.error('There is a BR tag inside the composition input!! This is really bad!');
+      }
+
+      if (isMount.current) {
+        el.innerHTML = normalizedHtml;
+        lastHtml.current = normalizedHtml;
+        isMount.current = false;
+      } else if (normalizedHtml !== normalizedCurrentHtml) {
+        el.innerHTML = normalizedHtml;
+        lastHtml.current = normalizedHtml;
+        handleChange();
+        replaceCaret(el);
+      }
+    }, [handleChange, html]);
+
+    return (
+      <div
+        {...rest}
+        ref={elRef}
+        // We don't want rich html editing
+        contentEditable={disabled ? false : 'plaintext-only'}
+        role="textbox"
+        spellCheck={true}
+        aria-disabled={disabled}
+        aria-multiline={true}
+        suppressContentEditableWarning
+        className={className}
+        style={style}
+        onInput={onInput}
+        onKeyUp={onKeyUp}
+        onKeyDown={_onKeyDown}
+        onCopy={onCopy}
+        onClick={_onClick}
+      >
+        {children}
+      </div>
+    );
+  }
+);
+
+const CompositionInput = styled(UnstyledCompositionInput)<{
+  placeholder?: string;
+  scrollbarPadding?: number;
+}>`
+  height: 100%;
+  width: 100%;
+  flex-grow: 1;
+  color: var(--text-primary-color);
+  font-family: var(--font-default);
+  line-height: var(--font-size-h2);
+  letter-spacing: inherit;
+  font-size: inherit;
+  white-space: pre-wrap;
+  overflow-wrap: break-word;
+  text-align: start;
+  max-height: 50vh;
+  min-height: 28px;
+  padding-top: 0;
+  padding-inline-end: ${props => props.scrollbarPadding ?? 0}px;
+  padding-inline-start: 0.375em;
+  padding-bottom: 0.25em;
+  overflow-y: auto;
+  scrollbar-gutter: stable;
+  outline: none;
+  border: none;
+  text-rendering: optimizeLegibility;
+
+  user-select: text;
+  * {
+    user-select: text;
+  }
+
+  &:empty:before {
+    content: attr(placeholder);
+    display: block;
+    color: #aaa;
```

Comment:
expected to be #aaa no matter the theme?

Reply:
fixed